src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int PRIME_NUMBER = 31;
+    private static final int PRIME_NUMBER1 = 311;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0 && table == null) {
+            threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = bucketIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else if (table[index].key == key
+                || key != null && table[index].key.equals(key)) {
+            table[index].value = newNode.value;
+            size--;
+        } else if (table[index].next == null) {
+            table[index].next = newNode;
+        } else {
+            Node<K, V> node = table[index];
+            while (node.next != null) {
+                node = node.next;
+            }
+            node.next = newNode;
+        }
+        size++;
+        if (size == threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0 || table == null) {
+            return null;
+        }
+        int index = bucketIndex(key);
+        while (table[index] != null) {
+            if (table[index].key == key
+                    || key != null && table[index].key.equals(key)) {
+                return table[index].value;
+            }
+            table[index] = table[index].next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private Node<K, V> next;
+        private V value;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
```

Comment:
If you don't use this field, it is better to remove it

Reply:
how can I remove it if I need it for the Node constructor and creating new nodes later?